### PR TITLE
Godot 4.2 support

### PR DESCRIPTION
**Pull request by bitbrain**
This pull request makes this extension compatible with Godot 4.2 - please read comments for more details.

- also updated to latest GUT

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -57,7 +57,7 @@ jobs:
             fmod-studio-platform-folder: linux/studio/lib/x86_64
             fmod-library-suffix: so
             godot-executable-download-suffix: linux.x86_64.zip
-            godot-executable: Godot_v4.1.2-stable_linux.x86_64
+            godot-executable: Godot_v$GODOT_VERSION-stable_linux.x86_64
```
Comment (bitbrain) [inline .github/workflows/release.yml:60]: Standardising this, so one only has to ever bump the env var

File: godot-cpp
Comment (bitbrain) [inline godot-cpp:1]: Updated from `godot-4.1.2-stable` to `godot-4.2-stable`

File: demo/run_tests.sh [hunk 2]
```diff
@@ -1,18 +1,36 @@
...
+        if echo "$line" | grep -q "Passing"; then
+            passing=$(echo "$line" | awk '{print $2}')
+        fi
+    done
+
+    if [[ "$tests" -eq 0 ]]; then
+       echo "ERROR: No tests were found."
+       exit 1
     fi
-    if echo "$line" | grep -q "Failing tests     0"; then
```
Comment (bitbrain) [inline demo/run_tests.sh:8]: This does not work any longer in latest GUT. A better way is to extract the number of passing tests vs total tests and ensure they are the same.

**Review by CedNaru — APPROVED** [review]

**Review by piiertho — APPROVED** [review]
Thanks a lot for this work !  
Please set a meaningful commit message when squashing.

**Pull request merged**